### PR TITLE
Migrate inductor and torchbench workflows to start experimenting with a100 on aws

### PR DESCRIPTION
**Pull request by jeanschmidt**
Excluding nightly workflows, as they are more critical and run less frequently.

### CODE REVIEW

**Comment by jeanschmidt** [comment]
* `inductor-micro-benchmark / cuda12.1-py3.10-gcc9-sm80` is broken on base.
* `torchbench / cuda12.1-py3.10-gcc9-sm80 / test (torchbench_gcp_smoketest` is also broken, for months now.

So I guess we should ignore those failures, as both seems unrelated to the runner.

File: .github/workflows/inductor-cu124.yml [hunk 1]
```diff
@@ -18,8 +18,8 @@ concurrency:
 permissions: read-all
 
 jobs:
-  get-label-type:
-    name: get-label-type
+  get-default-label-prefix:
```
Comment (malfet) [inline .github/workflows/inductor-cu124.yml:21]: This is unrelated, isn't it?

**Review by malfet — APPROVED** [review]
LGTM, but consider moving change that renames `get-label-type` to `get-default-label-prefix` to a different PR (and also, why not call it `get-runner-prefix`?)

File: .github/workflows/inductor-cu124.yml (hunk 1 above)
Comment (ZainRizvi) [reply .github/workflows/inductor-cu124.yml:21]: This is giving the old runner determinator invocation a more distinctive name from the new runner determinator invocation

File: .github/workflows/inductor-periodic.yml [hunk 2]
```diff
@@ -28,12 +28,23 @@ jobs:
       curr_branch: ${{ github.head_ref || github.ref_name }}
       curr_ref_type: ${{ github.ref_type }}
 
+  get-a100-test-label-type:
+    name: get-a100-test-label-type
+    uses: pytorch/pytorch/.github/workflows/_runner-determinator.yml@main
+    if: github.event_name != 'schedule' || github.repository == 'pytorch/pytorch'
```
Comment (ZainRizvi) [inline .github/workflows/inductor-periodic.yml:34]: Also

```suggestion
    if: github.event_name != 'schedule' && github.repository == 'pytorch/pytorch'
```

File: .github/workflows/torchbench.yml [hunk 3]
```diff
@@ -11,28 +11,41 @@ concurrency:
...
     with:
       triggering_actor: ${{ github.triggering_actor }}
       issue_owner: ${{ github.event.pull_request.user.login || github.event.issue.user.login }}
       curr_branch: ${{ github.head_ref || github.ref_name }}
       curr_ref_type: ${{ github.ref_type }}
 
+  get-a100-test-label-type:
+    name: get-a100-test-label-type
+    uses: pytorch/pytorch/.github/workflows/_runner-determinator.yml@main
+    if: github.event_name != 'schedule' || github.repository == 'pytorch/pytorch'
```
Comment (ZainRizvi) [inline .github/workflows/torchbench.yml:27]:
```suggestion
    if: github.event_name != 'schedule' && github.repository == 'pytorch/pytorch'
```

**Review by ZainRizvi — APPROVED** [review]
The conditional seems buggy, but otherwise LGTM

**Comment by huydhn** [comment]
> * `inductor-micro-benchmark / cuda12.1-py3.10-gcc9-sm80` is broken on base.

Let me take a look a this, I completely missed this and didn't notice that only CPU benchmark results were showing on https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

> * `torchbench / cuda12.1-py3.10-gcc9-sm80 / test (torchbench_gcp_smoketest` is also broken, for months now.

cc @kit1980 Could you check if this one is still in use?

File: .github/workflows/inductor-micro-benchmark.yml [hunk 4]
```diff
@@ -26,18 +26,31 @@ jobs:
       curr_branch: ${{ github.head_ref || github.ref_name }}
       curr_ref_type: ${{ github.ref_type }}
 
+  get-a100-test-label-type:
+    name: get-a100-test-label-type
+    uses: pytorch/pytorch/.github/workflows/_runner-determinator.yml@main
+    if: github.event_name != 'schedule' || github.repository == 'pytorch/pytorch'
```
[the remark this answers is not in the document]
Comment (huydhn) [reply .github/workflows/inductor-micro-benchmark.yml:32]: I think `||` is correct, this is agate to only run this workflow with a schedule on `pytorch/pytorch`, otherwise, it will be run daily on canary

**Review by huydhn — APPROVED** [review]
LGTM! Let me take a look at the inductor micro benchmark failure

File: .github/workflows/inductor-cu124.yml (hunk 1 above)
Comment (jeanschmidt) [reply .github/workflows/inductor-cu124.yml:21]: @ZainRizvi requested this change to be included with the latest migration. In order to make naming better to understand, I guess it could be done in multiple PRs, but not sure about the benefits honestly :)

File: .github/workflows/inductor-cu124.yml [hunk 5]
```diff
@@ -28,13 +28,24 @@ jobs:
       curr_branch: ${{ github.head_ref || github.ref_name }}
       curr_ref_type: ${{ github.ref_type }}
 
+  get-a100-test-label-type:
+    name: get-a100-test-label-type
+    uses: pytorch/pytorch/.github/workflows/_runner-determinator.yml@main
+    if: github.event_name != 'schedule' || github.repository == 'pytorch/pytorch'
```
[the remark this answers is not in the document]
Comment (jeanschmidt) [reply .github/workflows/inductor-cu124.yml:34]: I don't know, not sure why this check is there, just copied the conditions that are in the old `get-label-type` now `get-default-label-prefix` job.
